source/material_model/visco_plastic.cc

```diff
@@ -158,50 +158,73 @@ namespace aspect
               }
             }
 
-          double phi = angles_internal_friction[j];
-          double coh = cohesions[j];
+          std::pair<double, double> yield_parameters (cohesions[j], angles_internal_friction[j]);
```

Comment:
This logic needs to be repeated in `fill_plastic_outputs` as @naliboff found. Instead of repeating the whole block we should put it into a new function of the following type:
```
template <int dim>
std::array<double, 3>
ViscoPlastic::compute_weakened_yield_parameters(const double viscosity_pre_yield)
```
which returns cohesion, internal friction angle, and modified viscosity. This way we remove the duplication and avoid future bugs like the one we had to search for this time.

Reply:
@naliboff made some changes on his branch, but I think not quite what you suggest here, and I think he also rebased to a more current ASPECT version. Do you just want me to force-push all of his changes to this pull request, or should we close this one and @naliboff opens a new one?

Reply:
Actually, I don't think you can, because my aspect is not forked from geodynamics/aspect. But I am happy to do it once your done. Whatever seems easier for you.

Reply:
@jdannberg After I finish the change @gassmoeller and I discussed today and update the tests, I'll rebase again to the current master. If you also rebase to the current master on your branch I think you should be able pull in the commits from my branch directly? Maybe there is a cleaner way?

Reply:
@jdannberg @gassmoeller - ok, my branch (below) has a new commit and was rebased a minute ok to the current ASPECT master. @jdannberg - can you try pulling in my two commits or rebasing relative to my branch?
https://github.com/naliboff/aspect/tree/strain_weakening_logic

Hopefully all that is left is updating the tests and doing a final check on documentation/logic.

Reply:
Thanks! I just took your branch and force-pushed all of the changes. If you still have to make changes and don't rebase any more, I will simply be able to pull your changes to my branch.

Reply:
@jdannberg - Doh, I forgot to run astyle. I just added that commit to my branch.

Reply:
@jdannberg @gassmoeller - Hmm, three tests failing, but two appear to be due to small changes in screen-output. The third test actually has a small change to the solution. I guess we need a final look at that test (visco_plastic_yield_strain_weakening_compositions) to see if something is off with the new logic.